src/backup.py

```diff
@@ -7,13 +7,18 @@
 from typing import Any, Optional
 
 import boto3
+from botocore.config import Config
 from botocore.exceptions import BotoCoreError, ClientError
 from botocore.exceptions import ConnectionError as BotoConnectionError
 from pydantic import BaseModel, Field, validator
 
 logger = logging.getLogger(__name__)
 
 
+class S3Error(Exception):
+    """Generic S3 Exception."""
+
+
 class S3Parameters(BaseModel):
```

Comment:
Not blocker: But I've just realized that this reminds me of a discussion once @gregory-schiano 
Why is S3Parameters a class?

Reply:
That's a good question.

Initially, following the [contributing guide](https://github.com/canonical/is-charms-contributing-guide), a NamedTuple or similar could be worth just to group arguments. S3Parameters will not only be used for the S3Client, but in next PRs it will be also used for the workload (to set environment variables and to use as arguments in exec commands to the container).

However, as in SynapseConfig or ProxyConfig, it is using `pydantic` for some validations/parsing, and that's the reason it is not a NamedTuple. I think splitting that entity into two (one for parsing/validation) and another for internal use in backup is not worth for this simple case, and where S3 is a stable protocol.

Reply:
Hmm...Lets see how it goes, closing since is not a blocker